File: app/src/main/java/com/monk/aidldemo/binder/IPersonInterface.java
package com.monk.aidldemo.binder;

import android.os.IInterface;

import com.monk.aidldemo.bean.Person;

import java.util.List;

/**
 * 定义一个远程服务接口
 * @author monk
 * @date 2019-01-15
 */
public interface IPersonInterface extends IInterface {
    String DESCRIPTOR = "com.monk.aidldemo.binder";

    int TRANSACTION_addPerson = (android.os.IBinder.FIRST_CALL_TRANSACTION + 0);

    int TRANSACTION_getPersonList = (android.os.IBinder.FIRST_CALL_TRANSACTION + 1);

    void addPerson(Person p) throws android.os.RemoteException;

    List<Person> getPersonList() throws android.os.RemoteException;
}

File: module_fragment/src/main/java/com/monk/modulefragment/aidl/IPersonInterface.kt
package com.monk.modulefragment.aidl

import android.os.IBinder
import android.os.IInterface
import android.os.RemoteException
import com.monk.modulefragment.aidl.bean.Person

/**
 * ① 定义一个远程服务接口，继承自 IInterface
 *
 * @author monk
 * @date 2019-01-15
 */
interface IPersonInterface : IInterface {

    companion object {
        const val DESCRIPTOR  = "com.monk.modulefragment.aidl.IAidlInterface"
        const val TRANSACTION_addPerson = IBinder.FIRST_CALL_TRANSACTION + 0
        const val TRANSACTION_getPersonList = IBinder.FIRST_CALL_TRANSACTION + 1
    }

    @Throws(RemoteException::class)
    fun addPerson(p: Person?)

//    @Throws(RemoteException::class)
//    fun getPersonList(): List<Person?>

    @get:Throws(RemoteException::class)
    val personList:List<Person>?


}